src/styleguide/components/s-api-mock-test.vue

```diff
@@ -0,0 +1,379 @@
+            </td>
+            <td>
+              {{ endpoint.path }}
+            </td>
+            <td>
+              <e-checkbox v-model="configurations[endpoint.header].enabled"
+                          name="debug-mode-enabled"
+                          value
+              />
+            </td>
+            <td>
+              <e-select v-model="configurations[endpoint.header].status"
+                        :options="statusOptions"
+                        name="status"
+              />
+            </td>
+            <td>
+              <input v-model="configurations[endpoint.header].response"
+                     type="text"
+                     name="response"
+              >
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  import { defineComponent } from 'vue';
+  import {
+    RequestHandler,
+    ResponseResolver,
+    rest,
+  } from 'msw';
+  import mockWorker from '@/styleguide/api/browser';
+  import eCheckbox from '@/components/e-checkbox.vue';
+  import eSelect from '@/components/e-select.vue';
+
+  interface IDebugConfiguration {
+    header: string;
+    enabled: boolean;
+    status: string;
+    response: string;
+  }
+
+  interface ISetup {
+    storageKey: string;
+    statusOptions: {
+      label: string;
+      value: string;
+    }[]
+  }
+
+  interface IData {
+    search: string;
+    handlers: ReadonlyArray<RequestHandler> | null;
+    configurations: {
+      [key: string]: IDebugConfiguration;
+    },
+  }
+
+  /**
+   * Lists all active Mock API Endpoint handlers and allows enabling error mode for them.
+   */
+  export default defineComponent({
+    name: 's-api-mock-test',
+    components: { eSelect, eCheckbox },
+
+    // props: {},
+    emits: {
+      close: null,
+    },
+
+    setup(): ISetup {
+      return {
+        storageKey: 'sApiMockTest',
+        statusOptions: [
+          {
+            label: '200 OK',
+            value: '200',
+          },
+          {
+            label: '400 Bad Request',
+            value: '400',
+          },
+          {
+            label: '403 Forbidden',
+            value: '403',
+          },
+          {
+            label: '500 Internal Server Error',
+            value: '500',
+          },
+        ],
+      };
+    },
+    data(): IData {
+      return {
+        search: '',
+        handlers: null,
+        configurations: {},
+      };
+    },
+
+    computed: {
+      /**
+       * Returns list of endpoints  (filtered by search).
+       */
+      filteredEndpoints(): object[] | null {
+        const { search, handlers } = this;
+
+        if (!handlers) {
+          return null;
+        }
+
+        const mappedHandlers = handlers.map(handler => handler.info);
+
+        if (!search) {
+          return mappedHandlers;
+        }
+
+        return mappedHandlers.filter(endpoint => endpoint.header.toLowerCase().includes(search.toLowerCase()));
```

Comment:
sidenote: it would be better, to execute the filtering in a separate computed. The way you implemented it now, the whole mapping has to be redone on every search query change.